content/doc/book/managing/nodes.adoc

```diff
@@ -14,6 +14,119 @@ endif::[]
 
 = Managing Nodes
 
+### Monitor and Restart Offline Slaves
+This script can monitor and restart offline nodes if they are not disconnected manually.
+
+
+Of course you can disable email notification. Thanks author for email notification.
+
+You can run this script in Jenkins Console. But I think it is good idea to create Jenkins task, which will be run periodically (for example hourly). Groovy Plugin needs for Jenkins jobs.
+
+Also see link:https://wiki.jenkins-ci.org/display/JENKINS/Display+Information+About+Nodes[Display Information About Nodes]
+
+
+[source,syntaxhighlighter-pre]
```

Comment:
Since this is source code written in the groovy language, the highlighting should be based on groovy syntax rather than `syntaxhighlighter-pre`.

```suggestion
[source,groovy]
```